lib/QoreProgram.cpp

```diff
@@ -165,6 +166,18 @@ void qore_program_private_base::setDefines() {
    }
 }
 
+void qore_program_private_base::checkDefines(ExceptionSink *xsink) {
+    dmap_t::iterator blp = dmap.find("PO_BROKEN_LOGIC_PRECEDENCE");
+    dmap_t::iterator blpw = dmap.find("PO_BROKEN_LOGIC_PRECEDENCE_WARN");
+    if (blp == dmap.end() || !blp->second || !blp->second->getAsBool()) {
+        if (blpw != dmap.end() && blpw->second && blpw->second->getAsBool()) {
+            if (xsink) {
+                xsink->raiseException("PARSE-DIRECTIVE-ERROR", "broken-logic-precedence-warn cannot be set without broken-logic-precedence");
```

Comment:
this actually should not be a parse option but rather a warning, which would be a different implementation, but would make more sense